added window icon support for wayland

SrS2225a opened the pull request:
Added support for SDR++ to show the correct window icon under wayland sessions, instead of the default wayland one. As you can see in this screenshot: 
![image](https://github.com/user-attachments/assets/9986063f-aad0-4861-9eb5-5753b8ab8380)

AlexandreRouma commented:
As indicated above, your changes fail to build on most linux distros... You should check that your code properly compiles with different library versions before opening a pull request. I guess there's a reason why (as stated in the contributing.md and PR template) I don't accept code pull requests, even a single line one manages to be broken...

parkerlreed commented:
And at least for Plasma Wayland the string has to be "sdrpp" as it must match the icon file basename to show up.

SrS2225a commented:
> As indicated below, your changes fail to build on most linux distros... You should check that your code properly compiles with different library versions before opening a pull request. I guess there's a reason why (as stated in the contributing.md and PR template) I don't accept code pull requests, even a single line one manages to be broken...

Thank you for bringing this to my attention! I initially focused on testing the functionality in my current environment, which uses the latest versions of the libraries. I didn't anticipate issues with older versions, as I assumed compatibility would be maintained across versions. Moving forward, I'll make sure to test the code on different library versions and fix any build issues before submitting an updated PR.

SrS2225a commented:
> And at least for Plasma Wayland the string has to be "sdrpp" as it must match the icon file basename to show up.

I am not really sure what you mean as I use plasma myself and it showed up fine with "SDR++", and not "sdrpp", as it should match the name of the .desktop file, maybe you got the names mixed up?

parkerlreed commented:
> > And at least for Plasma Wayland the string has to be "sdrpp" as it must match the icon file basename to show up.
> 
> I am not really sure what you mean as I use plasma myself and it showed up fine with "SDR++", and not "sdrpp", as it should match the name of the .desktop file, maybe you got the names mixed up?

I built your branch with the sdrpp-git AUR PKGBUILD and had no titlebar icon when it was "SDR++". Changing it to "sdrpp" worked.

I think I have the normal desktop file as provided by SDR++

```
[parker@parker-framework ~]$ cat /usr/share/applications/sdrpp.desktop 
[Desktop Entry]
Encoding=UTF-8
Version=1.0
Type=Application
Terminal=false
Exec=/usr/bin/sdrpp
Name=SDR++
Icon=/usr/share/sdrpp/icons/sdrpp.png
Categories=HamRadio
```

SrS2225a commented:
> > > And at least for Plasma Wayland the string has to be "sdrpp" as it must match the icon file basename to show up.
> > 
> > 
> > I am not really sure what you mean as I use plasma myself and it showed up fine with "SDR++", and not "sdrpp", as it should match the name of the .desktop file, maybe you got the names mixed up?
> 
> I built your branch with the sdrpp-git AUR PKGBUILD and had no titlebar icon when it was "SDR++". Changing it to "sdrpp" worked.
> 
> I think I have the normal desktop file as provided by SDR++. My testing confirmed my understanding of the behavior. The hint has to match the name of the .desktop file itself.
> 
> ```
> [parker@parker-framework ~]$ cat /usr/share/applications/sdrpp.desktop 
> [Desktop Entry]
> Encoding=UTF-8
> Version=1.0
> Type=Application
> Terminal=false
> Exec=/usr/bin/sdrpp
> Name=SDR++
> Icon=/usr/share/sdrpp/icons/sdrpp.png
> Categories=HamRadio
> ```

Thanks for clearing up the confusion Parker.  Wired how in my case it worked with "SDR++".  I will double check it. Thanks

SrS2225a commented:
Thanks for your patience. I have fixed all the mentioned issues so it should now build for older versions of glfw. The issue arises because the GLFW_WAYLAND_APP_ID macro is specific to newer versions of GLFW, which I only found out later. Thats why I added this check. Let me know if you need any further adjustments. Thanks!
